src/main/java/org/opentest4j/MultipleFailuresError.java

```diff
@@ -90,11 +92,13 @@ private static String pluralize(int count, String singular, String plural) {
 		return count == 1 ? singular : plural;
 	}
 
-	private static String nullSafeMessage(AssertionError failure) {
+	private static String nullSafeMessage(Throwable failure) {
+		if (failure == null) {
```

Comment:
> I'm personally of the opinion that an explicit `NullPointerException` should be thrown instead a-la Effective Java. What's your view on this @marcphilipp?

We've had lengthy discussions on NPE vs. IAE for `null` arguments before and I don't want to get into them again... I think in this case I could live with a NPE.

Regarding `TestExecutionListener`: I agree that not specifying a `Throwable` is not very helpful. However, it's a test engine may not be able to supply a `Throwable` in some cases. I think stopping the entire test run by throwing an exception would be a bit harsh in that case. We can continue the discussion in a separate issue, if need be. Let's get this one done, first. 🙂